Source/Core/Core/NetPlayClient.cpp

```diff
@@ -316,6 +318,8 @@ unsigned int NetPlayClient::OnData(sf::Packet& packet)
 
     INFO_LOG(NETPLAY, "Player %s (%d) left", m_players.find(pid)->second.name.c_str(), pid);
```

Comment:
We probably shouldn't be dereferencing this iterator without checking its validity first, especially since this is untrusted data from the network, and a malformed message could crash the application. The lookup also isn't guarded by the mutex. I can see that this is a pre-existing issue, but we should fix it while we're at it rather than increasing the chance of races.

I would suggest something like:

```cpp
  case NP_MSG_PLAYER_LEAVE:
  {
    PlayerId pid;
    packet >> pid;

    {
      std::lock_guard<std::recursive_mutex> lkp(m_crit.players);
      const auto it = m_players.find(pid);
      if (it == m_players.end())
        break;

      const auto& player = it->second;
      INFO_LOG(NETPLAY, "Player %s (%d) left", player.name.c_str(), pid);
      m_dialog->OnPlayerDisconnect(player.name);
      m_players.erase(m_players.find(pid));
    }

    m_dialog->Update();
   }
   break;
```

Otherwise, LGTM.